### PR TITLE
Account Service SIWF v2 URI Validation

### LINKED ISSUE #640: Account Service SIWF v2 URI Validation
Related: https://github.com/ProjectLibertyLabs/siwf/issues/200

Instead of validating just the domain, we should validate the URI. All of the examples should still work from the above issue.

### PR DESCRIPTION
**Pull request by mattheworris**
# Problem

Updates to support siwf v2.1.0

Closes #640 

# Solution

- Renamed `SIWF_V2_DOMAIN` to `SIWF_V2_URI_VALIDATION` and follow on config changes, tests, etc.
- Renamed `loginMsgDomain` to `loginMsgURIValidation`
- Updated `ENVIRONMENT.md`

### CODE REVIEW

File: apps/account-api/src/api.config.spec.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ describe('Account API Config', () => {
     GRAPH_ENVIRONMENT_TYPE: undefined,
     SIWF_URL: undefined,
     SIWF_V2_URL: undefined,
-    SIWF_V2_DOMAIN: undefined,
+    SIWF_V2_URI_VALIDATION: undefined,
```
Comment (saraswatpuneet) [inline apps/account-api/src/api.config.spec.ts:17]: will need an update in deployment/charts

File: apps/account-api/src/api.config.spec.ts [hunk 2]
```diff
@@ -75,6 +69,19 @@ describe('Account API Config', () => {
       expect(accountServiceConfig.siwfUrl).toStrictEqual(ALL_ENV.SIWF_URL);
     });
 
+    it('should get SIWF V2 URL', () => {
+      expect(accountServiceConfig.siwfV2Url).toStrictEqual(ALL_ENV.SIWF_V2_URL);
+    });
+
+    it.each([['https://example.com/login'], ['example://login'], ['localhost'], ['localhost:3030/login/path']])(
```
Comment (enddynayn) [inline apps/account-api/src/api.config.spec.ts:76]: nice!

**Review by enddynayn — APPROVED** [review]
LGTM 👍

File: apps/account-api/src/api.config.spec.ts [hunk 3]
```diff
@@ -41,12 +41,6 @@ describe('Account API Config', () => {
     it('invalid api timeout limit should fail', async () => shouldFailBadValues(ALL_ENV, 'API_TIMEOUT_MS', [0]));
 
     it('invalid url for SIWF_V2_URL', async () => shouldFailBadValues(ALL_ENV, 'SIWF_V2_URL', ['sdfdsf']));
-
-    it('invalid url SIWF_V2_DOMAIN', async () =>
-      shouldFailBadValues(ALL_ENV, 'SIWF_V2_DOMAIN', ['https://www.example.com']));
```
Comment (mattheworris) [inline apps/account-api/src/api.config.spec.ts:46]: These values are valid now.

File: deployment/k8s/frequency-gateway/templates/account.configmap.yaml [hunk 4]
```diff
@@ -18,7 +18,8 @@ data:
   HEALTH_CHECK_SUCCESS_THRESHOLD: {{ .Values.account.env.HEALTH_CHECK_SUCCESS_THRESHOLD | quote }}
   CAPACITY_LIMIT: {{ .Values.account.env.CAPACITY_LIMIT | quote }}
   SIWF_URL: {{ .Values.account.env.SIWF_URL | quote }}
-  SIWF_DOMAIN: {{ .Values.account.env.SIWF_DOMAIN | quote }}
+  SIWF_V2_URL: {{ .Values.account.env.SIWF_V2_URL | quote }}
```
Comment (mattheworris) [inline deployment/k8s/frequency-gateway/templates/account.configmap.yaml:21]: @saraswatpuneet This value is optional, not sure if it should be here?

**Pull request merged**